core/dbt/events/base_types.py

```diff
@@ -57,6 +56,9 @@ class Event(metaclass=ABCMeta):
     log_version: int = 1
     ts: Optional[datetime] = None  # use getter for non-optional
     pid: Optional[int] = None  # use getter for non-optional
+    report_node_data: Optional[Any] = None
+    node_status: Optional[str] = None
+    node_state: Optional[str] = None
```

Comment:
could this be modeled with a single class? If it's all or nothing, splitting out the optionals like this is a looser type fit. Something like this:
```python
node_info: Optional[NodeInfo]
```
where `NodeInfo` has report_node_data, node_status, and node_state.